src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -3784,8 +3784,10 @@ static class D
             Diagnostic(ErrorCode.ERR_InterceptorMustHaveMatchingThisParameter, @"InterceptsLocation(""Program.cs"", 10, 23)").WithArguments("ref Program this", "Program.InterceptableMethod()").WithLocation(16, 6));
     }
 
-    [Fact]
-    public void SignatureMismatch_11()
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/71714")]
+    [InlineData("ref readonly")]
```

Comment:
I'm wondering if we should warn when an rvalue is passed to a `ref readonly` parameter (we do that everywhere else except for interceptors it seems). The interceptor should just use `in` if it wants to intercept an rvalue receiver.

Reply:
I think if we do end up permitting this scenario, we actually shouldn't warn. We already do not warn, by design, for things like `[Obsolete]` on an interceptor. This seems to fall in a similar category.

Do you know if we consider the `this` parameter of a readonly struct method to be `in` or `ref readonly` on the language level? Or maybe it doesn't fall into that categorization at all?

It seems like the compiler treats such a `this` parameter as `in`, but I'm not sure of the full significance of that.

Reply:
> Do you know if we consider the `this` parameter of a readonly struct method to be `in` or `ref readonly` on the language level?
> It seems like the compiler treats such a `this` parameter as `in`

Yes, I think that's like an `in` parameter. It's fine to call readonly struct methods on rvalues, that would warn if they were considered `ref readonly`.

Reply:
> I think if we do end up permitting this scenario, we actually shouldn't warn. We already do not warn, by design, for things like `[Obsolete]` on an interceptor. This seems to fall in a similar category.

Makes sense to me, thanks.